crates/viewer/re_renderer/src/video/decoder/web.rs

```diff
@@ -268,6 +263,7 @@ impl VideoDecoder {
         chunk.set_duration(sample.duration.as_f64());
         let Some(chunk) = EncodedVideoChunk::new(&chunk)
             .inspect_err(|err| {
+                // TODO(#7373): return this error once the decoder tries to return a frame for this sample. how exactly?
```

Comment:
We may have to set the decoder into an "error state" if it fails to decode a frame, as there may not be a reasonable way to continue decoding. In that case we would just store the error, and close the underlying decoder.

The only reason why decoding should fail is if their video is incorrectly encoded or corrupted in some way. We could for example try again from the next keyframe, as long as the user requests it. So we'd store the _segment_ which is broken inside the error state, alongside the error message from the decoder, and present that to the user until they seek to a different segment.

Reply:
makes sense. I suppose until we have a concrete example of a video that runs into this case to experiment with we shouldn't go overboard with recovery strategies as we don't know how they will work out. But marking segments as broken seems like a good strategy 👍 :)

Reply:
I don't know if you've experienced this scenario: You're playing a video on the web, and it suddenly fails with an error of some kind. The player just stops, and the controls become inactive, so you can't seek to a different point in the video anymore. But then you reload the page, and seek after the point where the player previously stopped with an error, and the video continues playing back just fine.

I think we could do a lot better than web video players here, by always allowing seek even in the case of an error, and more importantly also highlight which segments are broken on the timeline, so the user knows exactly where to seek next to try and catch a part of the video which _can_ be played back.

This kind of behavior would definitely follow our "handle anything the user throws at it" philosophy

Reply:
(actually, I will turn this into a comment on the existing issue)